File: jenjin-core/src/main/java/com/binarymonks/jj/pools/managers/Vector2PoolManager.java
package com.binarymonks.jj.pools.managers;

import com.badlogic.gdx.math.Vector2;
import com.binarymonks.jj.pools.PoolManager;

public class Vector2PoolManager implements PoolManager<Vector2> {
    @Override
    public void reset(Vector2 vector2) {
        vector2.set(0, 0);
    }

    @Override
    public Vector2 create_new() {
        return new Vector2();
    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/pools/managers/Vector2PoolManager.kt
package com.binarymonks.jj.core.pools.managers

import com.badlogic.gdx.math.Vector2
import com.binarymonks.jj.core.pools.PoolManager

class Vector2PoolManager : PoolManager<Vector2> {
    override fun reset(vector2: Vector2) {
        vector2.set(0f, 0f)
    }

    override fun create_new(): Vector2 {
        return Vector2()
    }

    override fun dispose(vector2: Vector2) {

    }
}
